From 2e984715d77426b72d467ee3fb8c26658a7e4a26 Mon Sep 17 00:00:00 2001
From: Sandy Carter <bwrsandman@gmail.com>
Date: Sat, 30 Jan 2021 22:08:08 -0500
Subject: [PATCH 4/4] bgfx: make bx and bimg private

This makes it so in shared mode, bx and bimg static libs aren't installed
---
 cmake/bgfx.cmake     | 2 +-
 cmake/examples.cmake | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/bgfx.cmake b/cmake/bgfx.cmake
index c2757ad..a0a3d30 100755
--- a/cmake/bgfx.cmake
+++ b/cmake/bgfx.cmake
@@ -82,7 +82,7 @@ target_include_directories( bgfx
 		$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
 
 # bgfx depends on bx and bimg
-target_link_libraries( bgfx PUBLIC bx bimg )
+target_link_libraries( bgfx PRIVATE bx bimg )
 
 # ovr support
 if( BGFX_USE_OVR )
diff --git a/cmake/examples.cmake b/cmake/examples.cmake
index 68093b4..9076932 100755
--- a/cmake/examples.cmake
+++ b/cmake/examples.cmake
@@ -134,7 +134,7 @@ function( add_example ARG_NAME )
 	if( ARG_COMMON )
 		add_library( example-${ARG_NAME} STATIC EXCLUDE_FROM_ALL ${SOURCES} )
 		target_include_directories( example-${ARG_NAME} PUBLIC ${BGFX_DIR}/examples/common )
-		target_link_libraries( example-${ARG_NAME} PUBLIC bgfx dear-imgui meshoptimizer )
+		target_link_libraries( example-${ARG_NAME} PUBLIC bgfx bx bimg dear-imgui meshoptimizer )
 		if( BGFX_WITH_GLFW )
 			find_package( glfw3 REQUIRED )
 			target_link_libraries( example-${ARG_NAME} PUBLIC glfw )
-- 
2.30.0

